graph/input.go

```diff
@@ -81,24 +97,49 @@ func (ip *InputProcessor[T]) process1D(v []T, kernel *kernel.Kernel) error {
 		return err
 	}
 
-	switch ip.dtype {
-	case tensor.Float:
+	switch v := any(v).(type) {
```

Comment:
Generics is used for the input processor. Why the use of `any`? I think using ip.dtype will be more performant.

Reply:
any is interface{}

and ip.dtype wont work because of this

```
type Number interface {
	int32 | int64 | int | float32 | float64 | map[int64]float32 | map[string]float32 | map[string]int64 | map[int64][]byte | map[int64]float64 | map[string]float64
}

type InputProcessor[T Number] struct {
	index int
	shape []int
	dtype tensor.DataType
}
```

previously `type Numeber interface` was only `int32 | int64 | int | float32 | float64` and the switch case you were using does not account for the type specifically because this type can easily be casted to one another. then if you introduced new type like this `map[string]float32 | map[string]int64 | map[int64` then the Generic T inside the switch case breaks then you have to the casting manually in each of the  switch case.

Reply:
I get you. But we can have two types of processor:

InputProcessor
InputMapProcessor

We already run a type check in the `setInputs` function, we can just use that to our advantage by using the right processor for the types of inputs.

Instead of running multiple type checks, we just need to run one type check and use generics to our advantage.

Reply:
I like this approach. I will create another process for map. This should also fix the failing test

Reply:
@goodyduru for maptype, no dispatch function is used. This uses the type check already defined in the setInputs global scope, and casting MapType is complex, so the user input type when using map is enforced.

Reply:
Lgtm!